### ISSUE TITLE
List operations require pagination options but do not return pagination status

### ISSUE THREAD

**Issue by akarnani** [hashicorp/go-tfe#9]
For example the organizations `List` function (https://github.com/hashicorp/go-tfe/blob/master/organization.go#L96) accepts an `OrganizationListOptions` which is simply a wrapper around `ListOptions` which primarily specifies pagination options.  

This requires that consumers of the library manage pagination themselves.  However, no pagination information is returned from the `List` calls forcing callers to create hacks to ensure all pages have been retrieved.

I see two possible fixes:
1. Have the `List` calls handle pagination internally and return all possible items for each `List` call.
2. Each `List` call returns a struct which includes pagination info for the caller to utilize.  I utilized `int` pointers in the example below to account for the fact that occasionally certain pagination fields are `null` and wanted consistency among all fields.

```
type OrganizationsListResponse struct {
    Organizations []*Organization
    Pagination struct {
        CurrentPage *int
        NextPage *int
        PrevPage *int
        TotalCount *int
        TotalPages *int        
    }
}
```

**Comment by svanharmelen** [comment]
Thanks for the report @akarnani! We are currently working on adding pagination support (using more or less the second approach you also described here).

I will update (close) this issue when the support has been merged.

**Comment by svanharmelen** [comment]
This one is fixed by #12
